File: src/test/java/com/ject/studytrip/studylog/fixture/StudyLogFixture.java
package com.ject.studytrip.studylog.fixture;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.studylog.domain.factory.StudyLogFactory;
import com.ject.studytrip.studylog.domain.model.StudyLog;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import org.springframework.test.util.ReflectionTestUtils;

public class StudyLogFixture {
    private static final String STUDY_LOG_CONTENT = "TEST 학습 로그 내용";

    public static StudyLog createStudyLog(Member member, DailyGoal dailyGoal) {
        return StudyLogFactory.create(member, dailyGoal, STUDY_LOG_CONTENT);
    }

    public static StudyLog createStudyLogWithId(Long id, Member member, DailyGoal dailyGoal) {
        StudyLog studyLog = createStudyLog(member, dailyGoal);
        ReflectionTestUtils.setField(studyLog, "id", id);

        return studyLog;
    }
}

File: src/test/kotlin/com/ject/studytrip/studylog/fixture/StudyLogFixture.kt
package com.ject.studytrip.studylog.fixture

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.studylog.domain.factory.StudyLogFactory
import com.ject.studytrip.studylog.domain.model.StudyLog
import com.ject.studytrip.trip.domain.model.DailyGoal
import org.springframework.test.util.ReflectionTestUtils

class StudyLogFixture(
    private val member: Member,
    private val dailyGoal: DailyGoal,
) {
    var content: String = "TEST 학습 로그 내용"

    fun create(): StudyLog = StudyLogFactory.create(member, dailyGoal, content)

    fun createWithId(id: Long): StudyLog =
        create().also {
            ReflectionTestUtils.setField(it, "id", id)
        }
}
